### ISSUE TITLE
Full text score not working with projection

### ISSUE THREAD

**Issue by ctmay4** [MorphiaOrg/morphia#1412]
**Describe the bug**
I am unable to project the meta score field from a full text search and have it set in the resulting entity.  Note that I was having an issue with this same thing a while ago and reported it in #1243.  At that time the problem was I had the `score` field named incorrectly.  I fixed that and was getting the score.  Now it is not working.  I'm not sure how it broke (new MongoDB, new Java driver, new Morphia)?

**To Reproduce**

I have an entity which has a bunch of fields and also a transient `score` field:

```java
@Entity(value = "ndc", noClassnameStored = true)
@Indexes(@Index(fields = {@Field(value = "ndc", type = IndexType.TEXT), options = @IndexOptions(name = "text_index"))})
public class NdcProduct {

    @Id
    private ObjectId objectId;

    @Indexed
    @Property("ndc")
    private String ndc;

    // more fields are here and in the full text index

    @Transient
    private Double score;
}
```

I wrote a simple unit test to show the issue.  I am trying to go a fill text search and have the score be included on the entity.

```java
@Test
public void testScoredReturned() {
    Query<NdcProduct> q = _ds.createQuery(NdcProduct.class);
    q.search("Quinidine");
    q.project(Meta.textScore("score"));
    q.order(Meta.textScore("score"));

    System.out.println(q);

    List<NdcProduct> products = q.find().toList();
    assertThat(products).isNotEmpty();
    assertThat(products.get(0).getScore()).isNotNull();
}
```

The query is working and I believe the results are correct.  However the test fails on the last line because the `score` is null.  

The interesting thing is what is returned from the "println" above:

```
{ {"$text": {"$search": "Quinidine"}} , projection: {"score": {"$meta": "textScore"}} }
```

That will not even run in MongoDB:

```
> db.find({ {"$text": {"$search": "Quinidine"}} , projection: {"score": {"$meta": "textScore"}} })
2020-01-17T14:36:02.563-0500 E  QUERY    [js] uncaught exception: SyntaxError: expected property name, got '{' :
@(shell):1:10
```

It looks like the correct way to run this would be:

```
> db.ndc.find(
...    { $text: { $search: "daklinza" } },
...    { score: { $meta: "textScore" } }
... )
{ "_id" : ObjectId("577fa0a4f938a4d9b6cf5e24"), "ndc" : "0003-0215", "labeler" : "00003", "product" : "0215", "type_name" : "HUMAN PRESCRIPTION DRUG", "prop_name" : "DAKLINZA", "non_prop_name" : [ "daclatasvir" ], "dosage_form_name" : "TABLET", "route_name" : [ "ORAL" ], "start_marketing_date" : "20150727", "end_marketing_date" : "20191231", "marketing_category_name" : "NDA", "application_number" : "NDA206843", "labeler_name" : "E.R. Squibb & Sons, L.L.C.", "substances" : [ { "name" : "DACLATASVIR DIHYDROCHLORIDE", "strength" : "60", "unit" : "mg/1" } ], "pharm_class" : [ "Hepatitis C Virus NS5A Inhibitor [EPC]", "P-Glycoprotein Inhibitors [MoA]", "Organic Anion Transporting Polypeptide 1B1 Inhibitors [MoA]", "Organic Anion Transporting Polypeptide 1B3 Inhibitors [MoA]", "Breast Cancer Resistance Protein Inhibitors [MoA]" ], "packages" : [ { "package" : "01", "description" : "28 TABLET in 1 BOTTLE (0003-0215-01) ", "start_marketing_date" : "20150727", "end_marketing_date" : "20191231" } ], "date_added" : ISODate("2016-07-08T12:46:27.670Z"), "date_modified" : ISODate("2019-04-28T03:00:32.432Z"), "score" : 1.1 }
```

The other issue I am having is that this use to work sometime.  I'm not sure when it stopped (could be a while).  I tried rolling back to older versions of MongoDB and older versions of Morphia and I could still not get it working.

** Please complete the following information: **
 - Server Version: 4.2.1
 - Driver Version: 3.12.1
 - Morphia Version: 1.5.8

**Comment by ctmay4** [comment]
I think I found the issue.  When I updated Morphia at some point I was getting a deprecation warning for `@NotSaved`. The sourec code had no suggestion for what to use.  I assumed the replacement was `@Transient`.  If I switch it back to `@NotSaved` I get the score again.  

`@NotSaved` was removed in #1377.  What is the correct replacement for it?

**Comment by evanchooly** [comment]
Hrm.  I might have to resurrect that annotation, then.  Though maybe I might rename it to `@LoadOnly` as that makes a bit more sense.  thoughts?

**Comment by ctmay4** [comment]
I don't have a real preference.  `@LoadOnly` seems like a good choice.

**Comment by evanchooly** [comment]
OK.  I'll readd them both but deprecate `@NotSaved` in 2.0.

**Comment by evanchooly** [comment]
fixed in 2923f80d2955bfebb7ffef6a5d800d248de5dfcb
